### PR TITLE
add qrda encounter dedupe merge

### LINKED ISSUE #5364: Need to fix Merge and Duplicate.
### Describe the bug
Need to account for duplicate encounters by date.
I know, what a waste for something that simply does not occur in the real world!!!!

### PR DESCRIPTION
**Pull request by sjpadgett**
Fixes #5364 

#### Short description of what this resolves:
This will get us by with testing I believe. I'll know more when I run through all the test patients, again!

### CODE REVIEW

**Comment by sjpadgett** [comment]
image

File: interface/patient_file/merge_patients.php [hunk 1]
```diff
@@ -39,507 +39,583 @@
...
                         }
-                    } else {
-                        // we just delete the entry from the source.
-                        $sql = "DELETE FROM " . escape_table_name($tblname) .
-                        " WHERE " . escape_sql_column_name($colname, array($tblname))
-                        . " = ? AND `type` = ?";
-                        echo "<br />$sql";
+                        $sql = "UPDATE " . escape_table_name($table['encounter_table']) . " SET " .
+                            escape_sql_column_name($table['encounter_column'], array($table['encounter_table'])) . " = ? WHERE " .
+                            escape_sql_column_name($table['encounter_column'], array($table['encounter_table'])) . " = ?";
```
Comment (bradymiller) [inline interface/patient_file/merge_patients.php:333]: 2DV

File: interface/patient_file/merge_patients.php [hunk 2]
```diff
@@ -39,507 +39,583 @@
...
-                            logMergeEvent(
-                                $target_pid,
-                                "delete",
-                                "Deleted rows with " . $colname . " = " . $source_pid
-                                . " and type = " . $source_row['type'] .
-                                " in table " . $tblname
-                            );
+                            sqlStatement($sql, array($target, $source['encounter']));
+                            if ($GLOBALS['adodb']['db']->_connectionID->affected_rows) {
+                                echo "<br />$sql ($target)" . ($source['encounter']);
```
Comment (bradymiller) [inline interface/patient_file/merge_patients.php:338]: would prob html escape these 2 variables

**Comment by sjpadgett** [comment]
After these changes things are going much better. Deduplication has a few rules that I'll try to acknowledge in the UI:

image

**Comment by sjpadgett** [comment]
It took 4.5 minutes to test CMS127 with 20 patients start to finish. So each patient takes 1.8 sec import and maybe 10secs for cypress.

**Pull request merged**